### PR TITLE
Drop support for go1.12, and use errors.Is() for error-detection

### PR DESCRIPTION
**Pull request by thaJeztah**
follow-up to https://github.com/tonistiigi/fsutil/pull/81

### CODE REVIEW

File: walker.go [hunk 1]
```diff
@@ -219,10 +219,10 @@ func trimUntilIndex(str, sep string, count int) string {
...
 func isNotExist(err error) bool {
-	err = errors.Cause(err)
-	if os.IsNotExist(err) {
+	if errors.Is(err, os.ErrNotExist) || errors.Is(err, syscall.ENOTDIR) {
 		return true
 	}
+	err = errors.Cause(err)
 	if pe, ok := err.(*os.PathError); ok {
 		err = pe.Err
 	}
```
Comment (thaJeztah) [inline walker.go:228]: Wondering if this code is still needed,  or if the `errors.Is(err, syscall.ENOTDIR)` I added above captures this already

@AkihiroSuda @tonistiigi ?

**Comment by thaJeztah** [comment]
Oh! This doesn't work because travis still tests against go 1.12 as well

**Comment by thaJeztah** [comment]
```
The command "sudo ./test -test.v" exited with 0.
7.71s$ GOOS=darwin go build ./...
# github.com/tonistiigi/fsutil
./stat_unix.go:17:6: undefined: "github.com/pkg/errors".Is
./walker.go:222:5: undefined: "github.com/pkg/errors".Is
note: module requires Go 1.13
```

**Comment by thaJeztah** [comment]
I removed go1.12, and added go1.13, go1.14

If we want to continue support for go < 1.13, I can add `_go112` and `_go113` files to use the old implementation for older Go versions

File: walker.go (hunk 1 above)
Comment (tonistiigi) [reply walker.go:228]: This should use `errors.As` instead of `Cause`

Comment (tonistiigi) [reply walker.go:228]: Didn't see the line comparing to `syscall.ENOTDIR` below. I guess this part isn't needed at all then now.

Comment (thaJeztah) [reply walker.go:228]: Thanks! Yes, I was in doubt if `errors.Is()` would do everything that type-matching `err.(*os.PathError)` did. I'll remove the part

**Comment by thaJeztah** [comment]
Rebased; should be ready to go

**Pull request merged**